listenbrainz/webserver/views/playlist_api.py

```diff
@@ -444,10 +444,12 @@ def get_playlist(playlist_mbid):
     if playlist is None:
         raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
 
-    if not playlist.public:
-        user = validate_auth_header()
-        if playlist.creator_id != user["id"]:
-            raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
+    user = validate_auth_header(True)
+    user_id = None
+    if user:
+        user_id = user["id"]
+    if not playlist.is_visible_by(user_id):
```

Comment:
```suggestion
    user = validate_auth_header(True)
    if not playlist.is_visible_by(user["id"]):
```

Reply:
`validate_auth_header` returns None if the argument `optional=True` is set and no token is provided, so this will break.